src/utilities/focus.ts

```diff
@@ -131,13 +131,22 @@ export function getNextElement(
 }
 
 export function isElementVisible(element: HTMLElement): boolean {
-  return (
-    !!element &&
-    (element.offsetHeight !== 0 ||
-      element.offsetParent !== null ||
-      (element as any).isVisible === true || // used as a workaround for testing.
-      (element.getAttribute && element.getAttribute(IS_VISIBLE_ATTRIBUTE) === 'true'))
-  );
+  // If the element is not valid, return false.
+  if (!element || !element.getAttribute) {
+    return false;
+  }
+
+  let visibilityAttribute = element.getAttribute(IS_VISIBLE_ATTRIBUTE);
+
+  // If the element is explicitly marked with the visibility attribute, return that value as boolean.
+  if (visibilityAttribute !== null) {
```

Comment:
if(!visibilityAttribute) is not working correctly. Tests started failing in focus zone when I changed it.